haas/errors.py

```diff
@@ -80,7 +80,9 @@ class ProjectMismatchError(APIError):
     """An exception indicating that the resources given don't belong to the
     same project.
     """
-    status_code = 409  # Conflict
+    status_code = 412  # was 409. 412 is appropriate here.
```

Comment:
In general, don't describe the history in comments. This would be more reasonable thing to mention in a commit message.

Also, this really *isn't* an appropriate use of the 412 status code, which has a very specific use. From the url you supply below:

> The precondition **given in one or more of the request-header fields** evaluated to false when it was tested on the server. This response code allows the client to place preconditions on the current resource metainformation (header field data) and thus prevent the requested method from being applied to a resource other than the one intended.

(emphasis mine). It's part of a mechanism defined in:

https://tools.ietf.org/html/rfc7232